app/src/main/java/com/x8bit/bitwarden/data/platform/repository/SettingsRepositoryImpl.kt

```diff
@@ -530,6 +533,20 @@ class SettingsRepositoryImpl(
         settingsDiskSource.storeUseHasLoggedInPreviously(userId)
     }
 
+    /**
+     * If there isn't already one generated, generate a symmetric sync key that would be used
+     * for communicating via IPC.
+     */
+    private fun generateSymmetricSyncKeyIfNecessary() {
+        // If there is already an authenticator sync symmetric key, do nothing:
+        if (authDiskSource.authenticatorSyncSymmetricKey != null) {
+            return
+        }
+        // Otherwise, generate and store a key:
+        val secretKey = generateSecretKey().getOrNull() ?: return
```

Comment:
If the function returns here, is it possible for a user to be caught in some undesired state?

Reply:
Heh, good question... the short answer is yes, that _could_happen. 

I can't think of any reason generating a secret would fail. I wrapped it in a `Result` to be super dooper cautious, but ultimately its just calling `KeyGenerator` APIs that should fail, but technically can... 

So I guess, TLDR; I'm not worried about an accidental bad state here. This is just for super safety.

Reply:
I would think the only way this could actually throw is if `AES` encryption is not supported but [according to the docs](https://developer.android.com/reference/kotlin/javax/crypto/KeyGenerator) it should be supported on all devices since API 1. So something really weird would need to be happening for this to fail.